scripts/check-public-api.ts

```diff
@@ -247,28 +222,62 @@ export function parseBarrelFile(fileContent: string, regex: RegExp): string[] {
   return flatten(groups.map(group => group.split(',')));
 }
 
+function checkInternalReExports(fileContent: string, filePath: string): void {
+  const internalReExports = parseBarrelFile(
+    fileContent,
+    /\{([\w,]+)\}from'.*\/internal'/g
+  );
+  if (internalReExports.length) {
+    throw new Error(
+      `Re-exporting internal modules is not allowed. ${internalReExports
+        .map(reExport => `'${reExport}'`)
+        .join(', ')} exported in '${filePath}'.`
+    );
+  }
+}
+
+export async function parseIndexFile(filePath: string): Promise<string[]> {
+  const cwd = dirname(filePath);
+  const fileContent = await readFile(filePath, 'utf-8');
+  checkInternalReExports(fileContent, filePath);
+  const localExports = parseBarrelFile(fileContent, /\{([\w,]+)\}from'\./g);
```

Comment:
I believe the regex here can be assigned to const `regexExportedIndex` declared above and the variable used here instead